### ISSUE TITLE
Failed to install from operatorhub

### ISSUE THREAD

**Issue by lgarciaaco** [syndesisio/syndesis#5925]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description

Currently fuse-online is installed via https://github.com/syndesisio/fuse-online-install, which takes care that the ImageStreams are present with the correct version: https://github.com/syndesisio/fuse-online-install/blob/d0190189b85ad0a43e93cea0beb48326e14e8867/install_ocp.sh#L925.

For the operatorhub integration, we have to move this logic (creating ImageStreams) to the operator.

After discussing with @nicolaferraro and @heiko-braun, it seems that the best solution for this release would be to give the user the possibility to download the complete `syndesis-template.yaml` from the fuse-online repository.

**Comment by heiko-braun** [comment]
@lgarciaaco @nicolaferraro is this in progress, or fixed?

**Comment by lgarciaaco** [comment]
Fixed by https://github.com/syndesisio/syndesis/pull/5926

**Comment by lgarciaaco** [comment]
@pure-bot pls let me close this issue :)

**Comment by tmielke** [comment]
If you resolve this one, can you please also resolve https://issues.jboss.org/browse/ENTESB-10903?

**Comment by heiko-braun** [comment]
@lgarciaaco Items that require QE verification need to stay `Open` and will be move to the `Done` column.

**Comment by avano** [comment]
@lgarciaaco @nicolaferraro @heiko-braun adding a new parameter to the operator doesn't solve the issue that there are no image streams, as the new parameter points to the same teplate that is contained in the operator - with no image streams for syndesis infrastructure: https://raw.githubusercontent.com/syndesisio/fuse-online-install/1.7.4/resources/fuse-online-template.yml

Image streams are by design not present for productized templates: https://github.com/syndesisio/syndesis/blob/master/install/generator/02-syndesis-image-streams.yml.mustache#L1


We don't have the template with image streams, (I assume, didn't inspect it closely) the images are feeded from `install_ocp.sh` into https://github.com/syndesisio/fuse-online-install/blob/1.7.4/templates/fuse-online-image-streams.yml (as you can see, here are only the placeholders)

I followed the guide from Luis and I did this change to the upstream CSV file: https://github.com/avano/syndesis/commit/c7e8f7d7ef3970a4ce0ff566f4662f2d579ddeb3

And I ended up with no image streams:

```
oc get all -n avano    
NAME                                    READY   STATUS      RESTARTS   AGE
pod/broker-amq-1-deploy                 0/1     Completed   0          65m
pod/syndesis-operator-68fc4cf68-x9v8p   1/1     Running     0          66m

NAME                                 DESIRED   CURRENT   READY   AGE
replicationcontroller/broker-amq-1   0         0         0       65m

NAME                          TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)               AGE
service/broker-amq-tcp        ClusterIP   172.30.140.153   <none>        61616/TCP,61613/TCP   65m
service/komodo-server         ClusterIP   172.30.89.157    <none>        80/TCP                65m
service/syndesis-db           ClusterIP   172.30.209.117   <none>        5432/TCP              65m
service/syndesis-meta         ClusterIP   172.30.124.71    <none>        80/TCP                65m
service/syndesis-oauthproxy   ClusterIP   172.30.8.115     <none>        8443/TCP              65m
service/syndesis-prometheus   ClusterIP   172.30.17.162    <none>        80/TCP                65m
service/syndesis-server       ClusterIP   172.30.213.166   <none>        80/TCP                65m
service/syndesis-ui           ClusterIP   172.30.246.107   <none>        80/TCP                65m
service/todo                  ClusterIP   172.30.119.173   <none>        8080/TCP              65m

NAME                                READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/syndesis-operator   1/1     1            1           67m

NAME                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/syndesis-operator-68fc4cf68   1         1         1       67m

NAME                                                     REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfig.apps.openshift.io/broker-amq            1          0         0         config,image(jboss-amq-63:1.3)
deploymentconfig.apps.openshift.io/komodo-server         0          0         0         config,image(fuse-komodo-server:latest)
deploymentconfig.apps.openshift.io/syndesis-db           0          1         0         config,image(postgres_exporter:v0.4.7),image(postgresql:9.5)
deploymentconfig.apps.openshift.io/syndesis-meta         0          1         0         config,image(fuse-ignite-meta:1.7)
deploymentconfig.apps.openshift.io/syndesis-oauthproxy   0          1         0         config,image(oauth-proxy:v1.1.0)
deploymentconfig.apps.openshift.io/syndesis-prometheus   0          1         0         config,image(prometheus:v2.1.0)
deploymentconfig.apps.openshift.io/syndesis-server       0          1         0         config,image(fuse-ignite-server:1.7)
deploymentconfig.apps.openshift.io/syndesis-ui           0          1         0         config,image(fuse-ignite-ui:1.7)
deploymentconfig.apps.openshift.io/todo                  0          1         0         config,image(todo:latest)

NAME                                  TYPE     FROM   LATEST
buildconfig.build.openshift.io/todo   Source   Git    0

NAME                                          IMAGE REPOSITORY                                                      TAGS   UPDATED
imagestream.image.openshift.io/jboss-amq-63   image-registry.openshift-image-registry.svc:5000/avano/jboss-amq-63   1.3    About an hour ago
imagestream.image.openshift.io/todo           image-registry.openshift-image-registry.svc:5000/avano/todo                  

NAME                                HOST/PORT                                                         PATH   SERVICES              PORT   TERMINATION          WILDCARD
route.route.openshift.io/syndesis   syndesis-avano.apps.syndesis-qe.syndesis-qe.fw.rhcloud.com               syndesis-oauthproxy   8443   reencrypt/Redirect   None
route.route.openshift.io/todo       todo-syndesis-avano.apps.syndesis-qe.syndesis-qe.fw.rhcloud.com   /      todo                  8080   edge/Allow           None

```

**Comment by heiko-braun** [comment]
@avano please expain how you tried to install. Are you using the scripts in https://github.com/syndesisio/fuse-online-install/tree/1.7.4 ?

**Comment by avano** [comment]
@heiko-braun I was following the guide from Luis how to get the custom operator into operator hub in OCP4: https://docs.google.com/document/d/1CKdv1FacWLeZ8xvbv0NCqR_lKqQJ0WcILbm75iru1XA/edit# but I didn't create my own resources and instead of that I used the resources from syndesis repository. So what I did was:

1. Change this https://github.com/avano/syndesis/commit/c7e8f7d7ef3970a4ce0ff566f4662f2d579ddeb3
2. In `install/operator/deploy/manifests` I used `operator-courier --verbose push . avano syndesis 1.7.0 "$AUTH_TOKEN"` to push the bundle to the quay repository and changed that to be public
3. I created an operator source in `openshift-marketplace` namespace:

```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: avano
  namespace: openshift-marketplace
spec:
  type: appregistry
  endpoint: https://quay.io/cnr
  registryNamespace: avano
  displayName: "Syndesis operator"
  publisher: "avano"
```

4. I scaled down to zero `certified-operators` `community-operators` `redhat-operators` deployments, just to have only my operator in the operator hub tab in OCP4 console
5. I created the operator in my namespace using the operator hub in my namespace
6. I created a secret and linked it to syndesis-operator SA to be able to pull the image from quay (because the operator image I changed is in private repository)
7. I created the syndesis CR in the UI and operator started to spawn the infrastructure - but didn't actually spawn any new pod besides `broker-amq-1-deploy` because the image streams are missing
